src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -305,5 +405,77 @@ await _usersAndRolesDbContext.UserRoles
                         .OnSuccess(() => _userRoleService.AddGlobalRole(userId, roleId));
                 });
         }
+
+        private async Task<Either<ActionResult, Unit>> ValidateEmail(string email)
+        {
+            if (string.IsNullOrWhiteSpace(email))
+            {
+                return ValidationActionResult(InvalidEmailAddress);
+            }
+
+            return Unit.Instance;
+        }
+
+        private async Task<Either<ActionResult, Unit>> ValidateUserDoesNotExist(string email)
+        {
+            if (_usersAndRolesDbContext.Users.Any(u => u.Email.ToLower() == email.ToLower()))
+            {
+                return ValidationActionResult(UserAlreadyExists);
+            }
+
+            return Unit.Instance;
+        }
+
+        private async Task<Either<ActionResult, Publication>> ValidateReleaseIds(Publication publication,
+            List<Guid> releaseIds)
+        {
+            await _contentDbContext
+                .Entry(publication)
+                .Collection(p => p.Releases)
+                .LoadAsync();
+
+            var publicationReleaseIds = publication
+                .ListActiveReleases()
+                .Select(r => r.Id)
+                .ToList();
+
+            var intersectedReleaseIds = releaseIds.Intersect(publicationReleaseIds).ToList();
+
+            if (!intersectedReleaseIds.All(releaseIds.Contains)) // if lists don't contain same elements
+            {
+                return ValidationActionResult(NotAllReleasesBelongToPublication);
+            }
+
+            return publication;
+        }
+
+        private async Task<Either<ActionResult, Unit>> SendContributorInviteEmail(
+            string publicationTitle, List<Guid> releaseIds, string email)
+        {
+            var template = _configuration.GetValue<string>("NotifyContributorTemplateId");
+            var scheme = _httpContextAccessor.HttpContext.Request.Scheme;
+            var host = _httpContextAccessor.HttpContext.Request.Host;
+            var url = $"{scheme}://{host}/";
+
+            var releaseTitleBullets = await _contentDbContext.Releases
+                .AsAsyncEnumerable()
+                .Where(r => releaseIds.Contains(r.Id))
+                .Select(r => $"* {r.Title}")
+                .ToListAsync();
+            var releaseList = string.Join("\n", releaseTitleBullets);
```

Comment:
Can also use `releaseTitleBullets.JoinToString('\n')` here instead of `string.Join`

Reply:
I've made the change. Is there an advantage to doing it this way?

Reply:
In this example not really but in general extension methods can help remove some boiler plate code and complexity.
In this case there's little difference between `x.JoinToString('\n')` and the implementation of it, `string.Join('\n', x)`.

In a change I'm currently making to these extension methods I allow the collection to be of any type avoiding the need to 'ToString' all the items of the collection if they aren't already strings, so it's about to become slightly more useful.